src/esrs/state.rs

```diff
@@ -60,6 +62,24 @@ impl<S: Default> AggregateState<S> {
         }
     }
 
+    /// Consumes the aggregate state and generates a new one with the events applied to it,
+    /// as dictaded by `apply_event`.
+    pub fn apply_store_events<T, F>(self, store_events: Vec<StoreEvent<T>>, apply_event: F) -> Self
+    where
+        F: Fn(S, T) -> S,
+    {
+        store_events.into_iter().fold(self, |state, store_event| {
+            let sequence_number = *store_event.sequence_number();
+            let inner = apply_event(state.inner, store_event.payload);
+
+            Self {
+                sequence_number,
+                inner,
+                ..state
+            }
+        })
+    }
```

Comment:
Moved here from `AggregateManager`, as this really deals with the internals of AggregateState (which are encapsulated).
It needs the map `(State, Event) -> State` which really is `Aggregate::apply_event`.

I also removed the snippet
```let sequence_number: SequenceNumber = store_events.last().map_or(0, StoreEvent::sequence_number);```
that was appearing in the old `AggregateManager` implementation. Now this function works correctly even when passing an empty list of events, which correctly acts as noop.